Library/Homebrew/build.rb

```diff
@@ -128,7 +128,7 @@ def install
       else
         formula.prefix.mkpath
 
-        formula.install
+        formula.run_install
```

Comment:
It's possible, but ugly, with this general approach: like `run_test`, `install` is overridden by the DSL definitions, so we can't modify it directly, but we need a level of indirection to introduce the logging setup before calling it directly. We could avoid a it by adding that `with_logging` call at this point in `build.rb`, but I think that's the wrong place, because it should _always_ be called when `install` will be run, not just when it's done through this `build.rb` file. The way to avoid introducing a `run_install` is to make an implicit rule that "every place that calls `install` needs to wrap it with a `with_logging do` call", which seems worse than introducing a new method that does that. Introducing a `run_install` level formalizes that and puts it in the code structure, instead of doco which contributors may or may not have read.